src/compiler.js

```diff
@@ -688,17 +688,40 @@ var Compiler = Object.extend({
             v = this.tmpid();
             frame.set(node.name.value, v);
 
-            this.emitLine('var ' + len + ' = ' + arr + '.length;');
-            this.emitLine('for(var ' + i + '=0; ' + i + ' < ' + arr + '.length; ' +
-                          i + '++) {');
-            this.emitLine('var ' + v + ' = ' + arr + '[' + i + '];');
-            this.emitLine('frame.set("' + node.name.value + '", ' + v + ');');
+            this.emitLine('if(runtime.isArray(' + arr + ')) {'); {
+                this.emitLine('var ' + len + ' = ' + arr + '.length;');
+                this.emitLine('for(var ' + i + '=0; ' + i + ' < ' + arr + '.length; ' +
+                              i + '++) {');
+                this.emitLine('var ' + v + ' = ' + arr + '[' + i + '];');
+                this.emitLine('frame.set("' + node.name.value + '", ' + v + ');');
 
-            this.emitLoopBindings(node, arr, i, len);
+                this.emitLoopBindings(node, arr, i, len);
 
-            this.withScopedSyntax(function() {
-                this.compile(node.body, frame);
-            });
+                this.withScopedSyntax(function() {
+                    this.compile(node.body, frame);
+                });
+
+                this.emitLine('}');
+            }
+
+            this.emitLine('} else if(runtime.isSet(' + arr + ')) {'); {
+                v = this.tmpid();
+                frame.set(node.name.value, v);
+
+                this.emitLine(i + ' = -1;');
+                this.emitLine('var ' + len + ' = ' + arr + '.size;');
+                this.emitLine('for(var ' + v + ' of ' + arr + ') {');
```

Comment:
oh, never mind, i'm wrong of course. because this syntax is generated as part of the compiled template regardless of whether a set is actually ever passed in. so yeah... I'm not sure how we can actually do this. Is there a way to iterate over sets without using `for...of`? Can we just convert to an array first?